components/fetch-build-scan-data-cmdline-tool/src/main/java/com/gradle/enterprise/cli/FetchBuildValidationDataCommand.java

```diff
@@ -66,11 +67,10 @@ public Integer call() throws Exception {
         CustomValueNames customValueKeys = loadCustomValueKeys(customValueMappingFile);
 
         logStartFetchingBuildScans();
-        List<BuildValidationData> buildValidationData = new ArrayList<>();
-        for (int i = 0; i < buildScanUrls.size(); i++) {
-            BuildValidationData validationData = fetchBuildScanData(i, buildScanUrls.get(i), customValueKeys);
-            buildValidationData.add(validationData);
-        }
+        List<BuildValidationData> buildValidationData = builds.stream()
+            .map(build -> fetchBuildScanData(build.runNum(), build.buildScanUrl(), customValueKeys))
+            .collect(Collectors.toList());
```

Comment:
Nice.

Reply:
I think I would extract this block into a separate method that returns `List<BuildValidationData> buildValidationData`. The reason is to have the same level of abstraction in this section.

And the same method extraction further down with `buildValidationData.forEach(this::printRow);`

Reply:
✅ Done